rest-server/engines/src/main/scala/com/actionml/engines/scaffold/ScaffoldEngine.scala

```diff
@@ -122,10 +117,23 @@ class ScaffoldEngine() extends Engine() with JsonParser {
     logger.trace(s"Got a query JSON string: $json")
     parseAndValidate[GenericQuery](json).andThen { query =>
       // query ok if training group exists or group params are in the dataset
-      val result = algo.predict(query)
+      val result = algo.query(query)
       Valid(result.toJson)
     }
   }
 
 }
 
+object ScaffoldEngine {
+  def apply(json: String): ScaffoldEngine = {
+    val engine = new ScaffoldEngine()
+    engine.initAndGet(json)
+  }
+
+  // in case we don't want to use "apply", which is magically connected to the class's constructor
+  def createEngine(json: String): ScaffoldEngine = {
+    val engine = new ScaffoldEngine()
+    engine.initAndGet(json)
+  }
+
+}
```

Comment:
This is a simple experimental builder. It will trigger other factories for the Dataset and Algorithm later. For now we just need to have this callable from the string "com.actionml.engines.ScaffoldEngine"